lib/spack/spack/environment/environment.py

```diff
@@ -926,46 +929,54 @@ def included_config_scopes(self):
             # allow paths to contain spack config/environment variables, etc.
             config_path = substitute_path_variables(config_path)
 
-            # strip file URL prefix, if needed, to avoid unnecessary remote
-            # config processing for local files
-            config_path = config_path.replace("file://", "")
+            include_url = urllib.parse.urlparse(config_path)
 
-            if not os.path.exists(config_path):
+            # Transform file:// URLs to direct includes.
+            if include_url.scheme == "file":
+                config_path = urllib.request.url2pathname(include_url.path)
```

Comment:
What is an example of a case where `config_path` != `include_url.path` here?

Reply:
Is this needed for `windows`?

Reply:
Windows because `file:///c:/x` would give a path `/C:/x` which url2pathname transforms into `C:\\x`.

But also on Linux it decodes the url, e.g. `file:///hello%20world` -> `/hello world`.